inc/cartridgeitem.class.php

```diff
@@ -80,11 +80,12 @@ function getPostAdditionalInfosForName() {
 
    function cleanDBonPurge() {
 
-      $class = new Cartridge();
-      $class->cleanDBonItemDelete($this->getType(), $this->fields['id']);
-
-      $class = new CartridgeItem_PrinterModel();
-      $class->cleanDBonItemDelete($this->getType(), $this->fields['id']);
+      $this->deleteChildrenAndRelationsFromDb(
+         [
+            Cartridge::class,
+            CartridgeItem_PrinterModel::class,
+         ]
+      );
 
       $class = new Alert();
       $class->cleanDBonItemDelete($this->getType(), $this->fields['id']);
```

Comment:
Why this one has not been replaced?

Reply:
Presence of the `cleanDBonItemDelete` method is a naming coincidence as `Alert` class does not extends CommonDBConnexity.
Maybe it should, but I do not really know all impacts of using `CommonDBChild` as base class for `Alert`.

Reply:
Ok, I've seen quite similar cases in the PR that were an evidence. OK for me, do not change.